tests/integration/test_charm.py

```diff
@@ -25,31 +26,32 @@ async def test_build_and_deploy(ops_test: OpsTest):
     """
     # build and deploy charm from local source folder
     charm = await ops_test.build_charm(".")
-    resources = {"httpbin-image": METADATA["resources"]["httpbin-image"]["upstream-source"]}
-    await ops_test.model.deploy(charm, resources=resources, application_name=APP_NAME)
-
-    # issuing dummy update_status just to trigger an event
-    await ops_test.model.set_config({"update-status-hook-interval": "10s"})
-
+    await ops_test.model.deploy(
+        charm,
+        application_name=APP_NAME,
+    )
     await ops_test.model.wait_for_idle(
         apps=[APP_NAME],
-        status="active",
+        status="waiting",
         raise_on_blocked=True,
         timeout=1000,
     )
-    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "active"
 
-    # effectively disable the update status from firing
-    await ops_test.model.set_config({"update-status-hook-interval": "60m"})
+    assert ops_test.model.applications[APP_NAME].units[0].workload_status == "waiting"
 
 
 @pytest.mark.abort_on_fail
 async def test_application_is_up(ops_test: OpsTest):
+    """Test if the application is up."""
+    # TODO: Add database write/read test
     status = await ops_test.model.get_status()  # noqa: F821
     address = status["applications"][APP_NAME]["units"][f"{APP_NAME}/0"]["address"]
 
-    url = f"http://{address}"
+    test_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    target = (address, DEFAULT_PORT)
+
+    logger.info("Querying app open port at %s:%s", address, DEFAULT_PORT)
+    port_status = test_socket.connect_ex(target)
+    test_socket.close()
```

Comment:
testing socket connections seems a little bit unconventional to me. [tests](https://github.com/canonical/operator-libs-linux/blob/main/tests/integration/test_apt.py#L25) in the operator lib check if `which <installed_package>` runs successfully - i might lean towards this sort of test instead

Reply:
Test `which` will only return if bin is in the path, not if app is running. This will be something to modify for sure, but more in the path of write/read to the database.